sdks/java/nexmark/build.gradle

```diff
@@ -46,3 +46,14 @@ dependencies {
 test {
   jvmArgs "-da"
 }
+
+def nexmarkArgsProperty = "nexmark.args"
+
+task run (type: JavaExec) {
+  main = "org.apache.beam.sdk.nexmark.Main"
+  classpath = sourceSets.main.runtimeClasspath
```

Comment:
This will run on the preshaded artifacts.

Create another configuration similar to the Flink validates runner which contains nexmark `shadow` configuration.
https://github.com/apache/beam/blob/c2d674448611d0490804533b96c41ba05e071572/runners/flink/build.gradle#L81

Reply:
Nexmark is not a test suite but a main application. Not sure if that changes your comment.

I wanted to build a bundled artifact that included all transitive deps and was standalone. Needs one per runner, which is unfortunate but a natural consequence. The shaded configuration currently fails to include things like joda time when I switch it to `compile` so I abandoned the effort. I will resume it, actually, as I do like being able to run this way. Configuration per runner with dynamic classifier?

Separately, what I am doing here is just using JavaExec to launch the main class. It isn't a library but an application so shading should be a noop when executed via JavaExec, is that not so? So I figure the non-bundled run should be entirely unshaded to save needless build.

Reply:
Nice idea. Done.